main.go

```diff
@@ -245,9 +296,139 @@ func initMetrics() {
 	}()
 }
 
-// createHealthFile creates the file required by Kubernetes' probes.
-func createHealthFile() {
-	if _, err := os.Create(probesFilePath); err != nil {
-		l.Log.Fatalf(`Unable to create the "healthy" file for Kubernetes' probes: %s`, err)
+// markConsumerStarted marks that the consumer has started processing messages
+func (h *consumerHealthState) markConsumerStarted() {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	h.consumerStarted = true
+	h.lastMessageTime = time.Now()
+	l.Log.Info("Consumer started")
+}
+
+// recordMessageProcessed records that a message was processed from a specific partition
+func (h *consumerHealthState) recordMessageProcessed(partition int32, offset int64) {
+	now := time.Now()
+	h.mu.Lock()
+	h.lastMessageTime = now
+	h.partitionOffsets[partition] = offset
+	h.messagesProcessed++
+	partitionCount := len(h.partitionOffsets)
+	h.mu.Unlock()
+
+	lastMessageTimestampGauge.Set(float64(now.Unix()))
+	partitionsAssignedGauge.Set(float64(partitionCount))
+}
+
+// checkHealth evaluates the overall health based on Kafka consumer activity and Sources API availability
+func (h *consumerHealthState) checkHealth() bool {
+	h.mu.RLock()
+	defer h.mu.RUnlock()
+
+	if !h.consumerStarted {
+		return false
+	}
+
+	if timeSince := time.Since(h.lastMessageTime); timeSince > consumerStaleTimeout {
+		l.Log.Warnf("Consumer stale: %v since last message", timeSince)
+		return false
+	}
```

Comment:
If I'm not mistaken this would make the consumer to be unhealthy if we don't receive messages in X amount of time. If there are periods when we don't have incoming messages this could trigger false positives and have our pods restarted. Is this because of a limitation on the Kafka library? I'm curious if the `ReadLastOffsset` method from the Kafka library can be used to compare the committed and the latest offset, to see if effectively the consumer is blocked.